redbot/cogs/cleanup/cleanup.py

```diff
@@ -541,3 +541,25 @@ def check(m):
             await mass_purge(to_delete, channel)
         else:
             await slow_deletion(to_delete)
+
+    @cleanup.command(name="spam")
+    @commands.guild_only()
+    @commands.bot_has_permissions(manage_messages=True)
+    async def cleanup_spam(self, ctx: commands.Context, number: int = 50):
+        """Deletes duplicate messages in the channel from the last X messages. Default 50."""
+        msgs = []
+        spam = []
+        async for msg in ctx.channel.history(limit=number + 1):
+            c = "{0.author.id}-{0.content}".format(msg)
```

Comment:
This isn't the safest way to remove spam messages - there can be messages that don't include any content (e.g. attachment-only messages or messages with embeds) and this would result in the bot removing those messages.
There's no good way to compare attachments so unfortunately the best we can do if we want this to be safe is ignore such messages completely. And as for embeds, we could compare them using `embed.to_dict()`:
```suggestion
            if msg.attachments:
                continue
            c = (msg.author.id, msg.content, [e.to_dict() for e in msg.embeds])
```